### PR TITLE
fix(pipelines-audio): resolve narrative stripping edge cases and add comprehensive tests

### PR DESCRIPTION
**Pull request by 0xSelenicDove**
## Description

This PR provides a comprehensive overhaul of the streaming narrative stripping logic in `pipelines-audio`. It addresses the complex edge cases and regressions identified by Codex and Gemini-Code-Assist in #1702, ensuring that math operators, code literals, and Markdown syntax are preserved while narrative spans are accurately stripped.

### 1. Robust Heuristic for Angle Brackets (`< >`)
* **Bidirectional Detection:** Introduced `isProbablyAngleTag` as a dedicated helper. It uses both lookahead (checking for digits/spaces) and lookbehind (checking for preceding characters like `List<T>`) to distinguish between code generics/math and narrative tags.
* **Hyphenated & Alphanumeric Support:** Refined the heuristic to correctly identify and protect HTML-like tags (e.g., `<h2>`) and hyphenated component tags (e.g., `<my-tag>`) that might be split across streaming chunks.

### 2. Range-Based Parsing State Machine
* **Non-Destructive Scanning:** Shifted from a single-pass "strip-as-you-go" approach to an index-based range deletion method in `processNarrative`.
* **Leakage Prevention:** By identifying and removing only validated and closed ranges, we ensure that unmatched openers (e.g., `Version (beta` or `a * b`) no longer cause subsequent non-narrative text to be swallowed.

### 3. Dynamic Streaming Fallback
* **Unified Buffer Safeguard:** Implemented a dynamic `fallbackLimit`. While active narrative markers are being tracked, the buffer limit is increased to 800 characters to prevent premature flushing of descriptive spans, while maintaining a sensitive 200-character limit for normal text to minimize latency.

## Edge Cases Solved
* **Markdown vs. Narrative:** Properly distinguishes between Markdown bullets (`* item`) and narrative emphasis (`*sigh*`) using whitespace-sensitive state toggling.
* **CJK Bracket Support:** Restored full stripping support for full-width/Chinese brackets (`（ ）`, `【 】`) that were missing in previous iterations.
* **Literals Preservation:** Fixed a regression in `keepNarrativeText` mode where global regex previously corrupted literal bracket characters in math/code expressions.

## Tests Added
* **Bidirectional Math Detection:** Validates `x<y` (preserved) vs. `List<String>` (preserved) vs. `<sigh>` (stripped).
* **Split-Chunk Tags:** Ensures tags split across streaming boundaries (e.g., `<s ... mile>`) are correctly tracked as unclosed.
* **Nested Bracket Handling:** Verifies complex nesting like `[laughs (quietly)]`.
* **Markdown Ambiguity:** Confirms that trailing punctuation inside brackets (e.g., `[note.]`) and unclosed asterisk算式 are handled correctly.

### CODE REVIEW

**Comment by 0xSelenicDove** [comment]
@gemini-code-assist I've resolved all the Codex P1/P2 issues and updated the unit tests. Can you do a final sweep to see if I missed any documentation, potential type mismatches, or if the code could be cleaner?

**Comment by 0xSelenicDove** [comment]
@gemini-code-assist help me check the new fixed code

File: packages/pipelines-audio/src/processors/tts-chunker.ts [hunk 1]
```diff
@@ -241,21 +241,147 @@ export async function chunkEmitter(
...
+
+  // Lookahead: if followed by num, space or equals, not a label
+  if (nextChar && /[0-9\s=]/.test(nextChar))
+    return false
+
+  if (prevChar && (isUnicodeLetter(prevChar) || /\d/.test(prevChar))) {
+    // fix: check whether remainder is piefix with any keywords, or contains the whole keyword
+    const isLikelyNarrative = NARRATIVE_KEYWORDS.some(kw =>
+      (remainder.length > 1 && kw.startsWith(remainder)) || remainder.startsWith(kw),
+    )
```
[the remark this answers is not in the document]
Comment (0xSelenicDove) [reply packages/pipelines-audio/src/processors/tts-chunker.ts:292]: Acknowledged. This is a deliberate trade-off to prioritize code/math literal preservation (e.g., List<T>) and minimize streaming latency. Given the infrequency of non-spaced custom tags, we're treating this as an acceptable edge case.

**Comment by 0xSelenicDove** [comment]
@nekomeowww I think I'm done here, pls check! If you have suggestions just leave a comment, thx!

File: packages/pipelines-audio/src/processors/tts-chunker.ts [hunk 2]
```diff
@@ -241,21 +241,147 @@ export async function chunkEmitter(
...
+  'sigh',
+  'action',
+  'note',
+  'breath',
+  'giggle',
+  'whisper',
+  'cry',
+  'smile',
+  'thought',
+]
```
Comment (nekomeowww) [inline packages/pipelines-audio/src/processors/tts-chunker.ts:267]: Should this be configurable?

**Review by nekomeowww — APPROVED** [review]

File: packages/pipelines-audio/src/processors/tts-chunker.ts (hunk 2 above)
Comment (0xSelenicDove) [reply packages/pipelines-audio/src/processors/tts-chunker.ts:267]: I thought about it, but decided to keep it as a hardcoded 'sane default' for now.

The primary goal of this PR is to stabilize the stripping logic and handle streaming edge cases. Making it configurable would increase the API surface area and might introduce new logical conflicts (e.g., users adding very short keywords that break code literal protection). If we see a strong demand for custom tags in the future, we can easily expose this as an option then.

**Review by nekomeowww — APPROVED** [review]
